File: workspace-ui/src/services/roomApi.js
//백엔드 막아둠
// 백엔드 임시 차단용 mock roomApi.js

// export const API_BASE = ''

// export async function fetchRooms() {
//   return {
//     rooms: [
//       {
//         id: 'room-1',
//         roomName: '개발 회의',
//         name: '개발 회의',
//         color: '#3b82f6',
//         role: 'owner',
//         createdAt: '2026-01-01',
//       },
//       {
//         id: 'room-2',
//         roomName: '캡스톤 회의',
//         name: '캡스톤 회의',
//         color: '#10b981',
//         role: 'member',
//         createdAt: '2026-01-02',
//       },
//     ],
//   }
// }

// export async function createRoom(roomName, color = '#7c3aed', visibility = 'public') {
//   return {
//     id: `room-${Date.now()}`,
//     roomName,
//     name: roomName,
//     color,
//     visibility,
//     role: 'owner',
//     createdAt: new Date().toISOString(),
//   }
// }

// export async function fetchRoomSessions(roomName) {
//   return {
//     sessions: [],
//   }
// }

// export async function fetchAllCalendarEvents() {
//   return {
//     events: [
//       {
//         id: 'event-all-1',
//         title: '전체 일정 예시',
//         date: '2026-05-04',
//         startTime: '2026-05-04T14:00:00',
//         endTime: '2026-05-04T15:00:00',
//         roomName: '개발 회의',
//         channel: '개발 회의',
//         isPrivate: false,
//         color: '#3b82f6',
//       },
//     ],
//   }
// }

// export async function fetchCalendarEvents(roomName) {
//   return {
//     events: [
//       {
//         id: `event-${roomName}-1`,
//         title: `${roomName} 일정 예시`,
//         date: '2026-05-04',
//         startTime: '2026-05-04T14:00:00',
//         endTime: '2026-05-04T15:00:00',
//         roomName,
//         channel: roomName,
//         isPrivate: false,
//         color: '#3b82f6',
//       },
//     ],
//   }
// }

// export async function fetchRoomCalendar(roomName) {
//   return fetchCalendarEvents(roomName)
// }

// export async function createRoomCalendarEvent(roomName, event) {
//   return {
//     id: `event-${Date.now()}`,
//     roomName,
//     channel: roomName,
//     title: event.title,
//     date: event.startTime ? event.startTime.split('T')[0] : event.date,
//     startTime: event.startTime,
//     endTime: event.endTime,
//     description: event.description || '',
//     isPrivate: event.isPrivate || false,
//     color: '#3b82f6',
//   }
// }

// export async function fetchRoomMembers(roomName) {
//   return {
//     members: [
//       {
//         userId: 'user-1',
//         name: 'Dev User',
//         email: 'dev@test.com',
//         role: 'owner',
//       },
//     ],
//   }
// }

// export async function createInviteLink(roomName) {
//   return {
//     inviteUrl: `http://localhost:5173/invite/TEST123?room=${encodeURIComponent(roomName)}`,
//   }
// }

// export async function fetchInviteInfo(inviteCode) {
//   return {
//     roomName: '테스트 회의실',
//     inviter: 'Dev User',
//   }
// }

// export async function acceptInvite(inviteCode) {
//   return { ok: true }
// }

// export async function createRoomInvite(roomName) {
//   return { ok: true }
// }

// export async function deleteRoomCalendarEvent(roomName, eventId) {
//   return { ok: true, roomName, eventId }
// }

const API_BASE = import.meta.env.VITE_API_BASE_URL || '/api'

async function parseResponse(res) {
  const data = await res.json().catch(() => null)

  if (!res.ok) {
    const message = data?.detail || data?.message || '요청 처리 중 오류가 발생했습니다.'
    throw new Error(message)
  }

  return data
}

export async function fetchRooms() {
  const res = await fetch(`${API_BASE}/rooms`, {
    method: 'GET',
    credentials: 'include',
  })

  return parseResponse(res)
}

export async function createRoom(roomName) {
  const res = await fetch(`${API_BASE}/rooms`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    credentials: 'include',
    body: JSON.stringify({ roomName }),
  })

  return parseResponse(res)
}

export async function fetchRoomSessions(roomName, channelId = null) {
  let url = `${API_BASE}/rooms/${encodeURIComponent(roomName)}/sessions`
  if (channelId) {
    url += `?channel_id=${encodeURIComponent(channelId)}`
  }
  const res = await fetch(url, {
    method: 'GET',
    credentials: 'include',
  })

  return parseResponse(res)
}

export async function fetchRoomMembers(roomName) {
  const res = await fetch(`${API_BASE}/rooms/${encodeURIComponent(roomName)}/members`, {
    method: 'GET',
    credentials: 'include',
  })

  return parseResponse(res)
}

export async function createInviteLink(roomName) {
  const res = await fetch(`${API_BASE}/rooms/${encodeURIComponent(roomName)}/invite-link`, {
    method: 'POST',
    credentials: 'include',
  })

  return parseResponse(res)
}

export async function fetchInviteInfo(inviteCode) {
  const res = await fetch(`${API_BASE}/rooms/invite/${encodeURIComponent(inviteCode)}`, {
    method: 'GET',
    credentials: 'include',
  })

  return parseResponse(res)
}

export async function acceptInvite(inviteCode) {
  const res = await fetch(`${API_BASE}/rooms/invite/accept`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    credentials: 'include',
    body: JSON.stringify({ inviteCode }),
  })

  return parseResponse(res)
}
export async function fetchCalendarEvents(roomName, channelId = null) {
  let url = `${API_BASE}/rooms/${encodeURIComponent(roomName)}/calendar/events`
  if (channelId) {
    url += `?channel_id=${encodeURIComponent(channelId)}`
  }
  const res = await fetch(url, {
    method: 'GET',
    credentials: 'include',
  })

  return parseResponse(res)
}

export async function createCalendarEvent(roomName, channelId, eventData) {
  let url = `${API_BASE}/rooms/${encodeURIComponent(roomName)}/calendar/events`
  if (channelId) {
    url += `?channel_id=${encodeURIComponent(channelId)}`
  }
  const res = await fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    credentials: 'include',
    body: JSON.stringify(eventData),
  })

  return parseResponse(res)
}

export async function deleteCalendarEvent(roomName, eventId) {
  const res = await fetch(`${API_BASE}/rooms/${encodeURIComponent(roomName)}/calendar/events/${encodeURIComponent(eventId)}`, {
    method: 'DELETE',
    credentials: 'include',
  })
  return parseResponse(res)
}

export async function fetchChannels(roomName) {
  const res = await fetch(`${API_BASE}/rooms/${encodeURIComponent(roomName)}/channels`, {
    method: 'GET',
    credentials: 'include',
  })
  return parseResponse(res)
}

export async function createChannel(roomName, channelName, description = '', color = '') {
  const res = await fetch(`${API_BASE}/rooms/${encodeURIComponent(roomName)}/channels`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    credentials: 'include',
    body: JSON.stringify({ channelName, description, color }),
  })
  return parseResponse(res)
}

export { API_BASE }

File: backend/room_api.py
import os
import sqlite3
import uuid
from datetime import datetime
from typing import Optional

from fastapi import APIRouter, HTTPException, Request
from pydantic import BaseModel

from storage_paths import DATA_DIR, get_room_sessions_dir, sanitize_room_name
from auth_api import get_login_user

router = APIRouter(prefix="/rooms", tags=["Rooms"])

DB_PATH = DATA_DIR / "meeting_app.sqlite3"


def conn():
    DATA_DIR.mkdir(parents=True, exist_ok=True)
    c = sqlite3.connect(DB_PATH)
    c.row_factory = sqlite3.Row
    return c


def ensure_room_tables():
    c = conn()
    cur = c.cursor()

##------------------추가-------------------##

    cur.execute(
    """
    CREATE TABLE IF NOT EXISTS calendar_events (
        id TEXT PRIMARY KEY,
        room_name TEXT NOT NULL,
        channel_id TEXT,
        title TEXT NOT NULL,
        start_time TEXT NOT NULL,
        end_time TEXT,
        description TEXT,
        is_private INTEGER DEFAULT 0,
        color TEXT,
        created_by TEXT NOT NULL,
        created_at TEXT NOT NULL
    )
    """
)
##------------------추가-------------------##

    cur.execute(
        """
        CREATE TABLE IF NOT EXISTS rooms (
            id TEXT PRIMARY KEY,
            room_name TEXT UNIQUE NOT NULL,
            owner_user_id TEXT NOT NULL,
            created_at TEXT NOT NULL
        )
        """
    )

    cur.execute(
        """
        CREATE TABLE IF NOT EXISTS channels (
            id TEXT PRIMARY KEY,
            room_name TEXT NOT NULL,
            channel_name TEXT NOT NULL,
            description TEXT,
            color TEXT,
            created_at TEXT NOT NULL,
            UNIQUE(room_name, channel_name)
        )
        """
    )

    cur.execute(
        """
        CREATE TABLE IF NOT EXISTS room_members (
            id TEXT PRIMARY KEY,
            room_name TEXT NOT NULL,
            user_id TEXT NOT NULL,
            role TEXT DEFAULT 'member',
            created_at TEXT NOT NULL,
            UNIQUE(room_name, user_id)
        )
        """
    )

    cur.execute(
        """
        CREATE TABLE IF NOT EXISTS room_invites (
            id TEXT PRIMARY KEY,
            room_name TEXT NOT NULL,
            inviter_user_id TEXT NOT NULL,
            invite_code TEXT UNIQUE NOT NULL,
            status TEXT DEFAULT 'active',
            created_at TEXT NOT NULL
        )
        """
    )

    cur.execute(
        """
        CREATE TABLE IF NOT EXISTS meeting_sessions (
            id TEXT PRIMARY KEY,
            room_name TEXT DEFAULT 'default_room',
            channel_id TEXT,
            title TEXT NOT NULL,
            meeting_time TEXT,
            keywords TEXT,
            meeting_type TEXT,
            realtime_recording_enabled INTEGER DEFAULT 1,
            created_at TEXT NOT NULL,
            stopped_at TEXT,
            status TEXT DEFAULT 'live'
        )
        """
    )

    cur.execute("PRAGMA table_info(meeting_sessions)")
    session_cols = {row[1] for row in cur.fetchall()}
    if "room_name" not in session_cols:
        cur.execute(
            "ALTER TABLE meeting_sessions ADD COLUMN room_name TEXT DEFAULT 'default_room'"
        )

    cur.execute("PRAGMA table_info(channels)")
    channel_cols = {row[1] for row in cur.fetchall()}
    if "color" not in channel_cols:
        cur.execute("ALTER TABLE channels ADD COLUMN color TEXT")

    c.commit()
    c.close()


class RoomCreatePayload(BaseModel):
    roomName: str

##------------------추가-------------------##

class CalendarEventPayload(BaseModel):
    title: str
    startTime: str
    endTime: str | None = None
    description: str | None = ""
    isPrivate: bool = False
    color: str | None = None

##------------------추가-------------------##

class InviteAcceptPayload(BaseModel):
    inviteCode: str

class ChannelCreatePayload(BaseModel):
    channelName: str
    description: str | None = ""
    color: str | None = None


def get_frontend_url():
    return os.getenv("FRONTEND_URL", "http://localhost:5173").rstrip("/")


def get_room_member(c: sqlite3.Connection, room_name: str, user_id: str):
    return c.execute(
        """
        SELECT *
        FROM room_members
        WHERE room_name = ? AND user_id = ?
        """,
        (room_name, user_id),
    ).fetchone()


def assert_room_member(c: sqlite3.Connection, room_name: str, user_id: str):
    member = get_room_member(c, room_name, user_id)
    if not member:
        raise HTTPException(status_code=403, detail="이 방에 접근할 권한이 없습니다.")
    return member


@router.get("")
def list_rooms(request: Request):
    ensure_room_tables()

    user = get_login_user(request)
    user_id = user["id"]

    c = conn()
    rows = c.execute(
        """
        SELECT r.*, m.role
        FROM rooms r
        JOIN room_members m ON r.room_name = m.room_name
        WHERE m.user_id = ?
        ORDER BY r.created_at DESC
        """,
        (user_id,),
    ).fetchall()
    c.close()

    return {
        "rooms": [
            {
                "id": row["id"],
                "roomName": row["room_name"],
                "ownerUserId": row["owner_user_id"],
                "role": row["role"],
                "createdAt": row["created_at"],
            }
            for row in rows
        ]
    }


@router.post("")
def create_room(payload: RoomCreatePayload, request: Request):
    ensure_room_tables()

    user = get_login_user(request)
    user_id = user["id"]

    room_name = sanitize_room_name(payload.roomName)
    now = datetime.now().isoformat()

    if not room_name:
        raise HTTPException(status_code=400, detail="방 이름이 비어 있습니다.")

    sessions_dir = get_room_sessions_dir(room_name)
    sessions_dir.mkdir(parents=True, exist_ok=True)

    c = conn()

    exists = c.execute(
        """
        SELECT id
        FROM rooms
        WHERE room_name = ?
        """,
        (room_name,),
    ).fetchone()

    if exists:
        c.close()
        raise HTTPException(status_code=409, detail="이미 존재하는 룸입니다.")

    room_id = str(uuid.uuid4())

    c.execute(
        """
        INSERT INTO rooms (id, room_name, owner_user_id, created_at)
        VALUES (?, ?, ?, ?)
        """,
        (room_id, room_name, user_id, now),
    )

    c.execute(
        """
        INSERT OR IGNORE INTO room_members (id, room_name, user_id, role, created_at)
        VALUES (?, ?, ?, ?, ?)
        """,
        (str(uuid.uuid4()), room_name, user_id, "owner", now),
    )

    c.commit()
    c.close()

    return {
        "id": room_id,
        "roomName": room_name,
        "ownerUserId": user_id,
        "role": "owner",
        "sessionsDir": str(sessions_dir),
        "createdAt": now,
    }


@router.get("/{room_name}/channels")
def list_room_channels(room_name: str, request: Request):
    ensure_room_tables()

    user = get_login_user(request)
    user_id = user["id"]
    safe_room_name = sanitize_room_name(room_name)

    c = conn()
    assert_room_member(c, safe_room_name, user_id)

    rows = c.execute(
        """
        SELECT *
        FROM channels
        WHERE room_name = ?
        ORDER BY created_at ASC
        """,
        (safe_room_name,),
    ).fetchall()
    c.close()

    return {
        "roomName": safe_room_name,
        "channels": [
            {
                "id": row["id"],
                "roomName": row["room_name"],
                "channelName": row["channel_name"],
                "description": row["description"],
                "color": row["color"],
                "createdAt": row["created_at"],
            }
            for row in rows
        ],
    }


@router.post("/{room_name}/channels")
def create_room_channel(room_name: str, payload: ChannelCreatePayload, request: Request):
    ensure_room_tables()

    user = get_login_user(request)
    user_id = user["id"]
    safe_room_name = sanitize_room_name(room_name)

    c = conn()
    assert_room_member(c, safe_room_name, user_id)

    channel_name = payload.channelName.strip()
    if not channel_name:
        c.close()
        raise HTTPException(status_code=400, detail="채널 이름이 비어 있습니다.")

    exists = c.execute(
        """
        SELECT id
        FROM channels
        WHERE room_name = ? AND channel_name = ?
        """,
        (safe_room_name, channel_name),
    ).fetchone()

    if exists:
        c.close()
        raise HTTPException(status_code=409, detail="이미 존재하는 채널 이름입니다.")

    channel_id = str(uuid.uuid4())
    now = datetime.now().isoformat()

    c.execute(
        """
        INSERT INTO channels (id, room_name, channel_name, description, color, created_at)
        VALUES (?, ?, ?, ?, ?, ?)
        """,
        (channel_id, safe_room_name, channel_name, payload.description, payload.color, now),
    )

    c.commit()
    c.close()

    return {
        "id": channel_id,
        "roomName": safe_room_name,
        "channelName": channel_name,
        "description": payload.description,
        "color": payload.color,
        "createdAt": now,
    }

@router.get("/{room_name}/sessions")
def list_room_sessions(room_name: str, request: Request, channel_id: Optional[str] = None):
    ensure_room_tables()

    user = get_login_user(request)
    user_id = user["id"]

    safe_room_name = sanitize_room_name(room_name)

    c = conn()

    assert_room_member(c, safe_room_name, user_id)

    if channel_id:
        rows = c.execute(
            """
            SELECT *
            FROM meeting_sessions
            WHERE room_name = ? AND channel_id = ?
            ORDER BY created_at DESC
            """,
            (safe_room_name, channel_id),
        ).fetchall()
    else:
        rows = c.execute(
            """
            SELECT *
            FROM meeting_sessions
            WHERE room_name = ?
            ORDER BY created_at DESC
            """,
            (safe_room_name,),
        ).fetchall()

    c.close()

    return {
        "roomName": safe_room_name,
        "sessions": [
            {
                "id": row["id"],
                "sessionId": row["id"],
                "roomName": row["room_name"],
                "channelId": row["channel_id"],
                "title": row["title"],
                "meetingType": row["meeting_type"],
                "meetingTime": row["meeting_time"],
                "keywords": row["keywords"],
                "status": row["status"],
                "createdAt": row["created_at"],
                "stoppedAt": row["stopped_at"],
            }
            for row in rows
        ],
    }


@router.get("/{room_name}/members")
def list_room_members(room_name: str, request: Request):
    ensure_room_tables()

    user = get_login_user(request)
    user_id = user["id"]

    safe_room_name = sanitize_room_name(room_name)

    c = conn()

    assert_room_member(c, safe_room_name, user_id)

    rows = c.execute(
        """
        SELECT m.room_name, m.user_id, m.role, m.created_at,
        u.email, u.name, u.picture
        FROM room_members m
        LEFT JOIN users u ON m.user_id = u.id
        WHERE m.room_name = ?
        ORDER BY
            CASE m.role
                WHEN 'owner' THEN 0
                ELSE 1
            END,
            m.created_at ASC
        """,
        (safe_room_name,),
    ).fetchall()

    c.close()

    return {
        "roomName": safe_room_name,
        "members": [
            {
                "userId": row["user_id"],
                "email": row["email"],
                "name": row["name"],
                "picture": row["picture"],
                "role": row["role"],
                "createdAt": row["created_at"],
            }
            for row in rows
        ],
    }


@router.post("/{room_name}/invite-link")
def create_invite_link(room_name: str, request: Request):
    ensure_room_tables()

    user = get_login_user(request)
    user_id = user["id"]

    safe_room_name = sanitize_room_name(room_name)
    now = datetime.now().isoformat()
    invite_code = str(uuid.uuid4())

    c = conn()

    member = assert_room_member(c, safe_room_name, user_id)

    if member["role"] not in ("owner", "admin", "member"):
        c.close()
        raise HTTPException(status_code=403, detail="초대 링크를 만들 권한이 없습니다.")

    room = c.execute(
        """
        SELECT *
        FROM rooms
        WHERE room_name = ?
        """,
        (safe_room_name,),
    ).fetchone()

    if not room:
        c.close()
        raise HTTPException(status_code=404, detail="존재하지 않는 방입니다.")

    c.execute(
        """
        INSERT INTO room_invites (
            id, room_name, inviter_user_id, invite_code, status, created_at
        )
        VALUES (?, ?, ?, ?, 'active', ?)
        """,
        (str(uuid.uuid4()), safe_room_name, user_id, invite_code, now),
    )

    c.commit()
    c.close()

    frontend_url = get_frontend_url()

    return {
        "roomName": safe_room_name,
        "inviteCode": invite_code,
        "inviteUrl": f"{frontend_url}/invite/{invite_code}",
        "createdAt": now,
    }


@router.get("/invite/{invite_code}")
def get_invite_info(invite_code: str):
    ensure_room_tables()

    c = conn()

    invite = c.execute(
        """
        SELECT i.*, r.id AS room_id, r.owner_user_id, r.created_at AS room_created_at,
               u.email AS inviter_email, u.name AS inviter_name, u.picture AS inviter_picture
        FROM room_invites i
        JOIN rooms r ON i.room_name = r.room_name
        LEFT JOIN users u ON i.inviter_user_id = u.id
        WHERE i.invite_code = ? AND i.status = 'active'
        """,
        (invite_code,),
    ).fetchone()

    c.close()

    if not invite:
        raise HTTPException(status_code=404, detail="유효하지 않은 초대 링크입니다.")

    return {
        "inviteCode": invite["invite_code"],
        "room": {
            "id": invite["room_id"],
            "roomName": invite["room_name"],
            "ownerUserId": invite["owner_user_id"],
            "createdAt": invite["room_created_at"],
        },
        "inviter": {
            "userId": invite["inviter_user_id"],
            "email": invite["inviter_email"],
            "name": invite["inviter_name"],
            "picture": invite["inviter_picture"],
        },
        "status": invite["status"],
        "createdAt": invite["created_at"],
    }


@router.post("/invite/accept")
def accept_invite(payload: InviteAcceptPayload, request: Request):
    ensure_room_tables()

    user = get_login_user(request)
    user_id = user["id"]

    now = datetime.now().isoformat()

    c = conn()

    invite = c.execute(
        """
        SELECT *
        FROM room_invites
        WHERE invite_code = ? AND status = 'active'
        """,
        (payload.inviteCode,),
    ).fetchone()

    if not invite:
        c.close()
        raise HTTPException(status_code=404, detail="유효하지 않은 초대 코드입니다.")

    room_name = invite["room_name"]

    room = c.execute(
        """
        SELECT *
        FROM rooms
        WHERE room_name = ?
        """,
        (room_name,),
    ).fetchone()

    if not room:
        c.close()
        raise HTTPException(status_code=404, detail="초대된 방이 존재하지 않습니다.")

    c.execute(
        """
        INSERT OR IGNORE INTO room_members (id, room_name, user_id, role, created_at)
        VALUES (?, ?, ?, 'member', ?)
        """,
        (str(uuid.uuid4()), room_name, user_id, now),
    )

    c.commit()
    c.close()

    return {
        "ok": True,
        "roomName": room_name,
        "userId": user_id,
        "role": "member",
    }


@router.get("/{room_name}/calendar/events")
def list_calendar_events(room_name: str, request: Request, channel_id: Optional[str] = None):
    ensure_room_tables()

    user = get_login_user(request)
    user_id = user["id"]

    safe_room_name = sanitize_room_name(room_name)

    c = conn()

    assert_room_member(c, safe_room_name, user_id)

    if channel_id:
        rows = c.execute(
            """
            SELECT *
            FROM calendar_events
            WHERE room_name = ? AND channel_id = ?
            ORDER BY start_time ASC
            """,
            (safe_room_name, channel_id),
        ).fetchall()
    else:
        rows = c.execute(
            """
            SELECT *
            FROM calendar_events
            WHERE room_name = ?
            ORDER BY start_time ASC
            """,
            (safe_room_name,),
        ).fetchall()

    c.close()

    return {
        "roomName": safe_room_name,
        "events": [
            {
                "id": row["id"],
                "roomName": row["room_name"],
                "channelId": row["channel_id"],
                "title": row["title"],
                "startTime": row["start_time"],
                "endTime": row["end_time"],
                "description": row["description"],
                "isPrivate": bool(row["is_private"]),
                "color": row["color"],
                "createdBy": row["created_by"],
                "createdAt": row["created_at"],
            }
            for row in rows
        ],
    }

##---------------------------추가---------------------------------##

@router.post("/{room_name}/calendar/events")
def create_calendar_event(
    room_name: str,
    payload: CalendarEventPayload,
    request: Request,
    channel_id: Optional[str] = None
):
    ensure_room_tables()

    user = get_login_user(request)
    user_id = user["id"]

    safe_room_name = sanitize_room_name(room_name)

    if not payload.title.strip():
        raise HTTPException(status_code=400, detail="일정 제목이 필요합니다.")

    now = datetime.now().isoformat()
    event_id = str(uuid.uuid4())

    c = conn()

    assert_room_member(c, safe_room_name, user_id)

    color = payload.color or "#3b82f6"

    c.execute(
        """
        INSERT INTO calendar_events (
            id, room_name, channel_id, title, start_time, end_time,
            description, is_private, color, created_by, created_at,
            scope, start_date, updated_at
        )
        VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
        """,
        (
            event_id,
            safe_room_name,
            channel_id,
            payload.title.strip(),
            payload.startTime,
            payload.endTime,
            payload.description or "",
            1 if payload.isPrivate else 0,
            color,
            user_id,
            now,
            "personal",
            payload.startTime.split("T")[0] if "T" in payload.startTime else payload.startTime,
            now,
        ),
    )

    c.commit()
    c.close()

    return {
        "id": event_id,
        "roomName": safe_room_name,
        "channelId": channel_id,
        "title": payload.title.strip(),
        "startTime": payload.startTime,
        "endTime": payload.endTime,
        "description": payload.description or "",
        "isPrivate": payload.isPrivate,
        "color": color,
        "createdBy": user_id,
        "createdAt": now,
    }

@router.delete("/{room_name}/calendar/events/{event_id}")
def delete_calendar_event(room_name: str, event_id: str, request: Request):
    ensure_room_tables()

    user = get_login_user(request)
    user_id = user["id"]

    safe_room_name = sanitize_room_name(room_name)

    c = conn()
    assert_room_member(c, safe_room_name, user_id)

    c.execute(
        """
        DELETE FROM calendar_events
        WHERE id = ? AND room_name = ?
        """,
        (event_id, safe_room_name),
    )

    c.commit()
    c.close()

    return {"ok": True, "eventId": event_id}